state/executor.go

```diff
@@ -667,6 +667,8 @@ func (t *Transition) apply(msg *types.Transaction) (*runtime.ExecutionResult, er
 		result = t.Call2(msg.From(), *(msg.To()), msg.Input(), value, gasLeft)
 	}
 
+	result.AccessList = initialAccessList
```

Comment:
Shouldn't we use `t.accessList` here, because it gets updated during the transaction execution?

```suggestion
	result.AccessList = t.accessList
```